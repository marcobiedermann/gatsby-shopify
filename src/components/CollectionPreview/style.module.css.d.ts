declare const styles: {
  readonly 'collectionPreview': string;
};
export = styles;
