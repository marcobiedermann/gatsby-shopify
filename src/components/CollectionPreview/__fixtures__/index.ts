import { CollectionPreviewProps } from '..';

const collectionPreview: CollectionPreviewProps = {
  handle: 'collection',
  image: {
    src: '',
  },
  shopifyId: '12345678-9012-3456-7890-123456789012',
  title: 'Collection Title',
};

export default collectionPreview;
