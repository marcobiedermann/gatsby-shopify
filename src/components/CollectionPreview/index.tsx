import { Link } from 'gatsby';
import React from 'react';
import Image from '../Image';
import { collectionPreview } from './style.module.css';

interface Source {
  srcSet: string;
  sizes: string;
  type: string;
}

type Layout = 'fixed' | 'fullWidth' | 'constrained';

interface Image {
  altText: string;
  gatsbyImageData: {
    images: {
      sources: Source[];
      fallback: {
        src: string;
        srcSet: string;
        sizes: string;
      };
    };
    layout: Layout;
    width: number;
    height: number;
  };
}

export interface CollectionPreviewProps {
  handle: string;
  image: Image;
  shopifyId: string;
  title: string;
}

function CollectionPreview(props: CollectionPreviewProps) {
  const { handle, image, title } = props;

  return (
    <div className={collectionPreview}>
      <figure>
        <Link to={`/collections/${handle}`}>
          <Image {...image} />
        </Link>
      </figure>
      <h3>
        <Link to={`/collections/${handle}`}>{title}</Link>
      </h3>
    </div>
  );
}

export default CollectionPreview;
