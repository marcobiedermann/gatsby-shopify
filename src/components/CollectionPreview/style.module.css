.collection-preview {
  padding: 2em;
}
