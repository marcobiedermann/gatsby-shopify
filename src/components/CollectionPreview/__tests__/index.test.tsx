import { render } from '@testing-library/react';
import React from 'react';
import CollectionPreview from '..';
import collectionPreview from '../__fixtures__';

describe('collection preview component', () => {
  it('should render correctly', () => {
    const { container } = render(<CollectionPreview {...collectionPreview} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
