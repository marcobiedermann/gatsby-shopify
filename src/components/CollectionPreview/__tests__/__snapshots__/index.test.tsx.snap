// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection preview component should render correctly 1`] = `
<div>
  <figure>
    <a
      href="/collections/collection"
    >
      <img
        alt="Collection Title"
        src=""
      />
    </a>
  </figure>
  <h3>
    <a
      href="/collections/collection"
    >
      Collection Title
    </a>
  </h3>
</div>
`;
